storage-access.bs

```diff
@@ -65,53 +65,87 @@ This specification depends on the Infra standard. [[!INFRA]]
+1. Return false.
+
+ISSUE: This should be prose, not an algorithm.
+
+A <dfn>third party context</dfn> is a [=/browsing context=] which is not a [=first party context=].
+
 <h3 id="ua-state">User Agent state related to storage access</h3>
 
-Each User Agent maintains a <dfn>storage access map</dfn>, an [=ordered map=] whose keys are [=partitioned storage keys=] and whose values are [=storage access flag sets=].
+A <dfn>storage access map</dfn> is a [=map=] whose keys are [=partitioned storage keys=] and whose values are [=storage access flag sets=].
 
-ISSUE: What's the lifetime of the [=storage access map=] and of its entries?
+User Agents maintain a single <dfn>global storage access map</dfn>.
 
-ISSUE(28): Tie the [=storage access map=] to agent clusters.
+ISSUE(2): What is the lifecycle of the [=global storage access map=]? How long do we remember its contents? Firefox and Safari differ here.
 
-A <dfn>partitioned storage key</dfn> is a [=pair=] consisting of a top-level site and a site, both [=sites=].
+Each [=agent cluster=] has a <dfn for="agent cluster">storage access map</dfn>.
 
-ISSUE: Replace this [=partitioned storage key=] concept with whatever Anne comes up with for [[!STORAGE]].
+When an [=agent cluster=] is created, its [=agent cluster/storage access map=] is initialized with a copy of the [=global storage access map=].
 
-A <dfn>storage access flag set</dfn> is a set of zero or more of the following flags, which are used to gate access to client-side storage for |site| when it's a [=third party=] on |top-level site|:
+To <dfn type="abstract-op">obtain the storage access map</dfn> for a [=top-level browsing context=] |context|, run the following steps:
 
-: The <dfn for="storage access flag set" id=has-storage-access-flag>has storage access flag</dfn>
-:: When set, this flag indicates |site| has access to its [=first party data=] when it's a [=third party=] on |top-level site|.
-: The <dfn for="storage access flag set" id=was-expressly-denied-storage-access-flag>was expressly denied storage access flag</dfn>
-:: When set, this flag indicates that the user expressly denied |site| access to its [=first party data=] when it's a [=third party=] on |top-level site|.
+1. Let |origin| be the [=Document/origin=] of |context|'s [=active document=].
+1. Let |key| be the result of [=obtain an agent cluster key|obtaining an agent cluster key=] given |origin|.
+1. Let |agent cluster| be the result of [=obtain a browsing context agent cluster|obtaining a browsing context agent cluster=] given |context|'s <a href="https://html.spec.whatwg.org/multipage/browsers.html#tlbc-group">group</a> and |key|
+1. Return |agent cluster|'s [=agent cluster/storage access map=].
+
+A <dfn>partitioned storage key</dfn> is a [=pair=] of [=sites=], <dfn for="partitioned storage key">top-level site</dfn>/<dfn for="partitioned storage key">embedded site</dfn>.
```

Comment:
If you don't mind, maybe use tuple here and the tuple syntax. I'm not entirely convinced we should keep pair as it has been confusing to folks.

Reply:
Fixed in 2312af4.